Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2722,6 +2728,148 @@ extension BrowserViewController: ContextMenuHelperDelegate {
     }
 }
 
+/**
+ A third party search engine Browser extension
+**/
+extension BrowserViewController {
+
+    /**
+    This allows us to find the part of the WKWebview that has access to
+    the inputAcessoryView
+    **/
+    func findWebViewFirstResponder(view:UIView) -> UIView? {
+        let subviews = view.subviews
+        if subviews.count == 0 {
+            return nil
+        }
+        for subview: UIView in subviews {
+            if subview.isFirstResponder() {
+                return subview
+            }
+            return findWebViewFirstResponder(subview)
+        }
+        return nil
+    }
+
+    func addCustomSearchButtonToWebView(webView:WKWebView) {
+        if self.customSearchEngineButton == nil {
+            self.customSearchEngineButton = self.searchEngineButton()
```

Comment:
Instead of a `nil` check that sets the value if `nil`, you can just make `customSearchEngineButton` a `lazy var` that will instantiate itself the first time it's accessed.